packages/connect-wallet-modal/src/components/Ledger/helpers.ts

```diff
@@ -0,0 +1,57 @@
+import { LS_KEY_DERIVATION_PATH } from '@reef-knot/ledger-connector';
+import { LedgerContextValue } from './LedgerContext';
+import { DERIVATION_PATHS } from './constants';
+
+export async function getTransport() {
+  return await (await import('@ledgerhq/hw-transport-webhid')).default.create();
+}
+
+export const saveLedgerDerivationPath = (path: string) => {
+  window?.localStorage.setItem(LS_KEY_DERIVATION_PATH, JSON.stringify(path));
+};
+
+export const loadLedgerDerivationPath = () => {
+  if (typeof window === 'undefined') {
+    return DERIVATION_PATHS[0].template;
+  }
+  return window.localStorage.getItem(LS_KEY_DERIVATION_PATH);
+};
+
+export const getDerivationPathsForPage = (
+  pathTemplate: string,
+  page: number,
+  perPage: number,
+) => {
+  const result: Record<string, string> = {};
+  Array.from({ length: perPage }).forEach((_, i) => {
+    const accountIndex = i + getFirstIndexOnPage(page, perPage);
+    result[accountIndex] = pathTemplate.replace('x', accountIndex.toString());
+  });
+  return result;
+};
+
+export const isDeviceBusy = (transport: LedgerContextValue['transport']) => {
+  // _appAPIlock indicates if a Ledger device is busy and returns the currently processed request's name
+  return transport.current?._appAPIlock;
```

Comment:
where did you get this prop? I cannot find it in any documentation for ledger

Reply:
I had to find it in the source code, here: https://github.com/LedgerHQ/ledger-live/blob/8f8480c3900b874c2504392c2cef864c1981c7c6/libs/ledgerjs/packages/hw-transport/src/Transport.ts#L356C2-L356C2

Reply:
You sure it will not break in a future? it is possible to use public API instead? looks a bit scary, at least give as much information as possible please

Reply:
@jabher
I can't be sure. Unfortunately, there is no public API for it, but I really need it. Because ledger works synchronously and throws errors if we bother it while it is busy. And our UI is obviously async and can send several requests at once.
Now I'm thinking, maybe we can catch this "device is busy" error and initiate another attempt in the "catch" block.

Reply:
OK, please leave as much details in comments as possible then, please, so if it will be broken it will be easier to fix

Reply:
refactored here: https://github.com/lidofinance/reef-knot/pull/85/commits/11ee0cada57163f6ee50570744e016e1a0b29864